src/app.ts

```diff
@@ -40,3 +48,92 @@ export default async function newApp(): Promise<Express> {
   }
   return app;
 }
+
+function asyncHandler(func: RequestHandler): RequestHandler {
+  return (req: Request, res: Response, next: NextFunction) =>
+    Promise.resolve(func(req, res, next)).catch(next);
```

Comment:
I don't understand this method, `Promise.resolve` will always resolve and `catch` will never be called (?)

Reply:
what you want is to have `func` to be an async `RequestHandler` (Express doesn't provide that type, you need to declare one yourself) - then:

```ts
func(req, res, next).catch(e => {
  log.error(e);
  res.status(500).json(...);
});
```